core/src/fgm/fgj/gamejamgame/GameEvent.java

```diff
@@ -173,131 +205,140 @@ private void handleRadiationEvent(SolarSystem solarSystem, Ship ship) {
 				return;
 			}
 		}
 		this.eventText = "Crew members seem happy.";
+		this.isPositive = false;
 	}
 
-	private String handleTradeEvent(Ship ship) {
+	private void handleTradeEvent() {
 		Random rand = new Random();
 		int giveResourceIndex = rand.nextInt(6);
 		int giveUnits = rand.nextInt(4) + 1;
 		String giveResource;
 		switch (giveResourceIndex) {
-			case 0: giveResource = "fuel"; ship.getCargoBay().increaseFuel(giveUnits); break;
-			case 1: giveResource = "metal"; ship.getCargoBay().increaseMetal(giveUnits); break;
-			case 2: giveResource = "water"; ship.getCargoBay().increaseWater(giveUnits); break;
-			case 3: giveResource = "gas"; ship.getCargoBay().increaseGas(giveUnits); break;
-			case 4: giveResource = "ammo"; ship.getCargoBay().increaseAmmo(giveUnits); break;
-			case 5: default: giveResource = "food"; ship.getCargoBay().increaseFood(giveUnits); break;
+			case 0: giveResource = "fuel"; this.ship.getCargoBay().increaseFuel(giveUnits); break;
+			case 1: giveResource = "metal"; this.ship.getCargoBay().increaseMetal(giveUnits); break;
+			case 2: giveResource = "water"; this.ship.getCargoBay().increaseWater(giveUnits); break;
+			case 3: giveResource = "gas"; this.ship.getCargoBay().increaseGas(giveUnits); break;
+			case 4: giveResource = "ammo"; this.ship.getCargoBay().increaseAmmo(giveUnits); break;
+			case 5: default: giveResource = "food"; this.ship.getCargoBay().increaseFood(giveUnits); break;
 		}
 		int receiveResourceIndex = rand.nextInt(6);
 		int receiveUnits = rand.nextInt(4) + 1;
 		String receiveResource;
 		switch (receiveResourceIndex) {
-			case 0: receiveResource = "food"; ship.getCargoBay().decreaseFuel(giveUnits); break;
-			case 1: receiveResource = "ammo"; ship.getCargoBay().decreaseMetal(giveUnits); break;
-			case 2: receiveResource = "gas"; ship.getCargoBay().decreaseWater(giveUnits); break;
-			case 3: receiveResource = "water"; ship.getCargoBay().decreaseGas(giveUnits); break;
-			case 4: receiveResource = "metal"; ship.getCargoBay().decreaseAmmo(giveUnits); break;
-			case 5: default: receiveResource = "fuel"; ship.getCargoBay().decreaseFood(giveUnits); break;
+			case 0: receiveResource = "food"; this.ship.getCargoBay().decreaseFuel(giveUnits); break;
+			case 1: receiveResource = "ammo"; this.ship.getCargoBay().decreaseMetal(giveUnits); break;
+			case 2: receiveResource = "gas"; this.ship.getCargoBay().decreaseWater(giveUnits); break;
+			case 3: receiveResource = "water"; this.ship.getCargoBay().decreaseGas(giveUnits); break;
+			case 4: receiveResource = "metal"; this.ship.getCargoBay().decreaseAmmo(giveUnits); break;
+			case 5: default: receiveResource = "fuel"; this.ship.getCargoBay().decreaseFood(giveUnits); break;
 		}
-		return "You gave them " + giveUnits + " " + giveResource + " and received " + receiveUnits + " " + receiveResource + ".";
+		if (this.planet != null) {
+			this.eventText = "You traded with the planet '" + this.planet.getName() + ".' ";
```

Comment:
I feel like there should be a boost to the planet's resources if this is the case.